frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,468 @@
+  return ko.kind;
+};
+
+const AttachStorageForm = () =>
+  class StorageFormComponent extends React.Component<
+    StorageProps,
+    StorageState
+    > {
+    constructor(props) {
+      super(props);
+      const existingResource = _.pick(props.obj, ['metadata', 'spec']);
+      const resourceObj = _.defaultsDeep({}, props.fixed, existingResource, {
+        apiVersion: 'v1',
+        data: {},
+        kind: props.kind,
+        metadata: {
+          name: '',
+        },
+        spec: {},
+      });
+
+      this.state = {
+        resourceObj,
+        inProgress: false,
+        formData: {},
+        readOnly: false,
+        disableForm: false,
+        checkboxItems: [
+          {
+            value: 'Mount as read-only',
+          },
+        ],
+        selectedOptions: [],
+      };
+    }
+
+    componentDidMount() {
+      const { kind, metadata } = this.state.resourceObj;
+      const ko = _.find(k8sModels, model => model.plural === kind);
+      const supportedKinds = [
+        'Deployment',
+        'DeploymentConfig',
+        'ReplicaSet',
+        'ReplicationController',
+      ];
+
+      if (!_.includes(supportedKinds, ko.kind)) {
+        this.setState({
+          error: `Storage is not supported for kind ${ko.kind}.`,
+          inProgress: false,
+        });
+        return;
+      }
+
+      // Get the current resource so we can add to its definition
+      k8sGet(ko, metadata.name, metadata.namespace).then(
+        obj => {
+          this.setState({ resourceObj: obj });
+        },
+        err => this.setState({ error: err.message })
+      );
+    }
+
+    onError = error => {
+      this.setState({
+        error,
+        inProgress: false,
+      });
+    };
+
+    handleChange = event => {
+      const name = event.target.name;
+      const value = event.target.value;
+      const formData = { ...this.state.formData, [name]: value };
+      this.setState({ formData });
+    };
+
+    onReadOnlyChanged = event => {
+      let readOnly = this.state.readOnly;
+      readOnly = readOnly ? false : true;
+      this.setState({ readOnly });
+    };
+
+    onFormDisable = disable => {
+      this.setState({
+        disableForm: disable,
+      });
+    };
+
+    isValidForm = () => {
+      // Check that a valid PVC has been choosen
+      if (!this.state.formData.claimName) {
+        this.setState({ error: 'Storage must be selected', inProgress: false });
+        return false;
+      }
+      return true;
+    };
+
+    // Add Attach storage //
+    save = (event: React.FormEvent<EventTarget>) => {
+      event.preventDefault();
+      if (!this.isValidForm()) {
+        return;
+      }
+      const { kind, metadata } = this.state.resourceObj;
+      const ko = _.find(k8sModels, model => model.kind === kind);
+      const resourceObj = { ...this.state.resourceObj };
+      const template = resourceObj.spec.template;
+      let { volumeName, mountPath, subPath, claimName } = this.state.formData;
```

Comment:
const

If we need to change one of these, let's declare it separately so the `const` values can be `const`. Or better, give it a different variable later.